common/src/chain/config/mod.rs

```diff
@@ -587,30 +588,74 @@ fn create_testnet_genesis() -> Genesis {
     )
 }
 
-pub fn regtest_genesis_values() -> (
+#[derive(Clone, Debug)]
+pub struct GenesisStakingSettings(String);
```

Comment:
What I meant is something that actually stores the settings in a struct, not just a mere wrapper over a string. The `FromStr` parses the settings string into the struct, including all the error checking and such. The command line parser will then be able to report format errors directly.

Reply:
Ah, ok. This is done, but I've forgotten to handle the different errors... I'll update this now.